src/main_service/main_service.rs

```diff
@@ -107,43 +160,42 @@ impl Service for MainService {
         info!("{}.run | Starting...", self.dbg_id);
         let dbg_id = self.dbg_id.clone();
         let exit = self.exit.clone();
+        let conf = self.conf.clone();
         info!("{}.run | Preparing thread...", dbg_id);
+
         let handle = thread::Builder::new().name(format!("{}.run", dbg_id.clone())).spawn(move || {
             let interval = 0.0; // from sampl_freq & buf_size
             let interval = Duration::from_secs_f64(interval);
             let mut cycle = ServiceCycle::new(&dbg_id, interval);
             let addr = "127.0.0.1:15181"; //other address in string format
+            let mut buf = Buffer::new(512);
 
-            let parent = "";
-            let conf_tree = ConfTree::new();
-            let config = MainService::new(parent, conf_tree);
-            for(freq, amp, phi) in config.signal{
-                let amplitude = amp;
-                let angle = phi;
-            }
-            let buf:[u8; 0] = [];
-            //let header = todo!("Udp message head, find detales here: https://github.com/a-givertzman/cma-server/issues/123#issue-2478437558");
             loop {
-                match Self::udp_bind(addr) {
-                    Ok(socket) => {
-                        cycle.start();
-                        // 
-                        // do the sampling actions
-                        // - calc amplitude
-                        // - add to buffer
-                        // - if buffer is full
-                        //      do the udp actions
-                        //      - build data splitting buffer amplitudes into bytes
-                        //      - build message using header and data
-                        //      - send message to the socket
-                        let bytes = buf.map(|amp: f64| {
-                            let amp: i16 = amp.round() as i16;
-                            let amp_bytes = amp.to_be_bytes();
-                        });
-                        socket.send(bytes);
-                        cycle.wait();
+                for (freq, amp, phi) in conf.signal.iter(){
+                    let amplitude = amp;
+                    match buf.add(*amplitude){
+                        Some(_) =>{
+                            match Self::udp_bind(addr){
+                                Ok(socket) => {
+                                    cycle.start();
+                                    let header = UdpHeader::new(0, 0, 32, 255);
```

Comment:
Replace literals with constants, defined in the responsible classes if possible.